src/mesh/fv_ops.cxx

```diff
@@ -1,10 +1,24 @@
-
 #include <bout/fv_ops.hxx>
 #include <bout/globals.hxx>
 #include <bout/msg_stack.hxx>
+#include <bout/output.hxx>
 #include <bout/utils.hxx>
 
-#include <bout/output.hxx>
+namespace {
+template <class T>
+struct Slices {
+  T c;
+  T up;
+  T down;
+  Slices(bool use_slices, const T& field)
+      : c(use_slices ? field : toFieldAligned(field)), up(use_slices ? field.yup() : c),
+        down(use_slices ? field.ydown() : c) {}
+};
+template <class T>
+Slices<T> makeslices(bool use_slices, const T& field) {
+  return Slices<T>(use_slices, field);
+}
```

Comment:
Just to note that we'll upgrade to C++17 soon and that should be able to deduce `T` without this helper